packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts

```diff
@@ -1224,22 +1224,40 @@ export class AmplifyAuthCognitoStack extends cdk.Stack implements AmplifyAuthCog
       this.deleteExistingHostedUIProviderCustomResource();
     }
 
-    if (props.hostedUIProviderCreds) {
-      JSON.parse(props.hostedUIProviderMeta).forEach(this.createHostedUIProviderResource);
+    if (props.hostedUIProviderMeta && props.hostedUIProviderCreds) {
+      const creds = JSON.parse(props.hostedUIProviderCreds);
+
+      JSON.parse(props.hostedUIProviderMeta).forEach((provider: ProviderMeta) => {
+        const providerCreds: ProviderCreds = creds.find(({ ProviderName }: ProviderCreds) => ProviderName === provider.ProviderName);
+        const hasProviderCreds = providerCreds.client_id && (providerCreds.client_secret || providerCreds.private_key);
+
+        this.createHostedUIProviderResource(provider, !!hasProviderCreds);
```

Comment:
```suggestion
        this.createHostedUIProviderResource(provider, hasProviderCreds);
```

Reply:
This makes sure that `hasProviderCreds` is always a boolean because `providerCreds.client_id && (providerCreds.client_secret || providerCreds.private_key)` might return `undefined`.